### ISSUE TITLE
corrupted rdb file or issue with redis-check-rdb - redisearch 2.0

### ISSUE THREAD

**Issue by raybog** [RediSearch/RediSearch#1549]
Starting fresh (empty)  Redisearch 2.0 (docker.io/redislabs/redisearch:latest) with following config:
```
dir /data
dbfilename redisearch.rdb
appendonly no
```
then running:
`redis-check-rdb redisearch.rdb `

getting:
```
[offset 0] Checking RDB file redisearch.rdb
[offset 26] AUX FIELD redis-ver = '6.0.5'
[offset 40] AUX FIELD redis-bits = '64'
[offset 52] AUX FIELD ctime = '1600300440'
[offset 67] AUX FIELD used-mem = '861840'
[offset 83] AUX FIELD aof-preamble = '0'
--- RDB ERROR DETECTED ---
[offset 84] Invalid object type: 247
[additional info] While doing: read-type
[additional info] Reading type 0 (string)
[info] 0 keys read
[info] 0 expires
[info] 0 already expired
```

Redis startup log:
```

1:C 16 Sep 2020 23:54:27.356 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 16 Sep 2020 23:54:27.356 # Redis version=6.0.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 16 Sep 2020 23:54:27.356 # Configuration loaded
1:M 16 Sep 2020 23:54:27.357 * Running mode=standalone, port=6379.
1:M 16 Sep 2020 23:54:27.357 # Server initialized
1:M 16 Sep 2020 23:54:27.358 * <search> Redis version found by RedisSearch : 6.0.5 - oss
1:M 16 Sep 2020 23:54:27.358 * <search> RediSearch version 2.0.0 (Git=v1.99.5-17-gd421035)
1:M 16 Sep 2020 23:54:27.358 * <search> Low level api version 1 initialized successfully
1:M 16 Sep 2020 23:54:27.358 * <search> concurrent writes: OFF, gc: ON, prefix min length: 1, prefix max expansions: 200, query timeout (ms): 500, timeout policy: return, cursor read size: 1000, cursor max idle (ms): 300000, max doctable size: 1000000, max number of search results:  1000000, search pool size: 20, index pool size: 8, 
1:M 16 Sep 2020 23:54:27.358 * <search> Initialized thread pool!
1:M 16 Sep 2020 23:54:27.358 * Module 'search' loaded from /usr/lib/redis/modules/redisearch.so
1:M 16 Sep 2020 23:54:27.359 * Loading RDB produced by version 6.0.5
1:M 16 Sep 2020 23:54:27.359 * RDB age 27 seconds
1:M 16 Sep 2020 23:54:27.359 * RDB memory usage when created 0.82 Mb
1:M 16 Sep 2020 23:54:27.359 * <module> Scanning all indexes
1:M 16 Sep 2020 23:54:27.359 * DB loaded from disk: 0.001 seconds
1:M 16 Sep 2020 23:54:27.359 * Ready to accept connections
1:M 16 Sep 2020 23:54:27.360 * <module> Scanning indexes in background
1:M 16 Sep 2020 23:54:27.360 * <module> Scanning indexes in background: done (scanned=0)

```

**Comment by ashtul** [comment]
@raybog
Thank you for bringing this up. 
Looking at the code of `redis-check-rdb`, it seems it does not handle `RDB_OPCODE_MODULE_AUX (247)` so this looks like a Redis issue.
We will sort it out but in the meanwhile, does redisearch perform as expected after loading from rdb, right?

**Comment by raybog** [comment]
@ashtul 
Thanks for looking at it.
I am not 100% sure but so far redisearch performs ok. I have just reported another issue #1568 but I guess those two are not related.

**Comment by ashtul** [comment]
fixed by redis/redis#7808

**Comment by jaylng** [comment]
How to solve this problem, I also encountered the same problem, redis version is 3.2.1, can be exported normally, but after exporting to run redis-check-rdb detection just exported this file will be this error:

[offset 0] Checking RDB file dump.rdb

[offset 26] AUX FIELD redis-ver = '3.2.1'

[offset 40] AUX FIELD redis-bits = '64'

[offset 52] AUX FIELD ctime = '1687745710'

[offset 67] AUX FIELD used-mem = '199983488'

[offset 69] Selecting DB ID 0

--- RDB ERROR DETECTED ---

[offset 23632] Invalid object type: 22

[additional info] While doing: read-type

[info] 96 keys read

[info] 0 expires

[info] 0 already expired

**Comment by oshadmi** [comment]
@jaylng Fixed in redis 6.0.9.
